jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/HTTP2Session.java

```diff
@@ -1946,15 +1946,19 @@ private boolean onIdleTimeout()
                         zeroStreamsAction = null;
                         failure = cause = new TimeoutException("Session idle timeout expired");
                     }
-                    default ->
-                    {
-                        if (LOG.isDebugEnabled())
-                            LOG.debug("Already closed, ignored idle timeout for {}", HTTP2Session.this);
-                        return false;
-                    }
+                    default -> terminate = true;
                 }
             }
 
+            if (terminate)
+            {
+                if (LOG.isDebugEnabled())
+                    LOG.debug("Already closed, ignored idle timeout for {}", HTTP2Session.this);
+                // Writes may be TCP congested, so termination never happened.
+                flusher.abort(new TimeoutException());
```

Comment:
All branches of the switch above result in a `new TimeoutException`, so just do it once before the switch

Reply:
Uhm, no. The `TimeoutException` is only allocated if `terminate=true` which only happens if the session is already in `CLOSED` state.

Reply:
It is also allocated in locally and remotely closed cases. At least give it the same message as the others.